src/ProcessedStory.ts

```diff
@@ -55,32 +56,39 @@ export function getDisabledRules(story: ProcessedStory): string[] {
   return story.parameters.axe.disabledRules;
 }
 
-function normalizeSkip(skipped?: unknown): boolean {
-  if (typeof skipped === 'undefined') {
-    return false;
-  }
-  if (typeof skipped !== 'boolean') {
-    throw new Error(`Value of 'skip' option '${skipped}' is invalid`);
-  }
-  return skipped;
+function normalizeSkip(skip: unknown, rawStory: StorybookStory) {
+  return parseWithFriendlyError(
+    () => Parameters.parseSkip(skip),
+    createInvalidParamErrorMessage(rawStory, 'skip'),
+  );
 }
 
-function normalizeDisabledRules(disabledRules?: unknown): string[] {
-  if (typeof disabledRules === 'undefined') {
-    return [];
-  }
-  if (!Array.isArray(disabledRules)) {
-    throw new Error(`Given disabledRules option '${JSON.stringify(disabledRules)}' is invalid`);
-  }
-  return disabledRules.map(String);
+function normalizeDisabledRules(disabledRules: unknown, rawStory: StorybookStory) {
+  return parseWithFriendlyError(
+    () => Parameters.parseDisabledRules(disabledRules),
+    createInvalidParamErrorMessage(rawStory, 'disabledRules'),
+  );
 }
 
-function normalizeWaitForSelector(waitForSelector?: unknown): string | undefined {
-  if (!waitForSelector) {
-    return undefined;
-  }
-  if (typeof waitForSelector !== 'string') {
-    throw new Error(`Value of 'waitForSelector' option '${waitForSelector}' is invalid`);
+function normalizeWaitForSelector(waitForSelector: unknown, rawStory: StorybookStory) {
+  return parseWithFriendlyError(
+    () => Parameters.parseWaitForSelector(waitForSelector),
+    createInvalidParamErrorMessage(rawStory, 'waitForSelector'),
+  );
+}
+
+function parseWithFriendlyError<T>(parser: () => T, errorMessage: string): T {
+  try {
+    return parser();
+  } catch (message) {
+    if (message instanceof Parameters.ParamError) {
+      throw new TypeError(errorMessage);
+    } else {
+      throw message;
+    }
```

Comment:
I couldn't find an easy way to pass an error message to `zodSchema.parse`, so we're stuck with try/catch.